### DISCUSSION TITLE
Non-response (or delayed response) to sendTransaction requests

### DISCUSSION THREAD

**Discussion by judeProground** [WalletConnect/walletconnect-monorepo#4290] [Developer Support]
Hello. We are requesting signatures from MetaMask using the react-native web3modal SDK.

1. Our dApp users often encounter situations where there is no response or the response is very slow (which we have to cancel the request) when requesting a signature. It seems that manually ending the WalletConnect session in the MetaMask app settings, or manually disconnecting from dApp(removing the storage value of the key "WALLETCONNECT_DEEPLINK_CHOICE"), and then reconnecting and requesting the signature again resolves the issue. Are there any known reasons why this might occur?

2. Alternatively, would providing the deeplink (or session information) used by users with a problem help to identify the cause?

Example:
- Deeplink
```
wc:84addae7275c8fc2139c7f85e4d3feecc4d360699395986562651c7d1233...@2?relay-protocol=irn&symKey=00e4b2fd61806a6f7f6d724e4de44a67a7cf60bb06c5cf5abde5b451d589...
```

- Session object (walletconnect provider.session)
```
{
  controller: 'c6074867d1890eb464200e81c604f2094ce31cb1094f562392bbe249a1ee6037'
  pairingTopic: '84addae7275c8fc2139c7f85e4d3feecc4d360699395986562651c7d1233efc8'
  // ...
}
```

It is difficult to create a reproducible example due to difficulties in accurately reproducing the problem.
Thank you for reading!

**Comment by judeProground** [comment]
In our current dApp, when we request sendTransaction while the MetaMask mobile app is closed (not running), MetaMask does not respond. The same issue persists even after subsequent requests.

**Comment by glitch-txs** [comment]
Hi, there are some issue with MetaMask currently, we're helping them to improve their WalletConnect implementation and fix them. This might take some time until the next update.

**Reply by judeProground** [reply]
Hi, @glitch-txs.

Was there any updates for those issues? Or do you have any information regarding the issues you mentioned? I'm curious to know if the situation I'm experiencing is related.

Thank you.

**Comment by MayankVHITS** [comment]
Hi @glitch-txs ,
We have received the same issue mentioned above. Kindly inform us if there are any updates available.
Thanks

**Comment by altmshfkgudtjr** [comment]
Hello, our app is currently experiencing the same phenomenon as above, and we are desperately looking forward to the stabilization of Metamask use through WalletConnect.

**Comment by korlaism** [comment]
Hello, we also have the same issue and we are desperately looking forward for a fix or any guide. Is there any other issue here or on metamask ( that you know of ) tracking this ?

**Reply by glitch-txs** [reply]
There's a fix coming from MetaMask in its next version, ETA end of April.
